lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1540,13 +1544,19 @@ createDispatchingDiagnosticConsumerIfNeeded(
           subconsumers.emplace_back(input.file(), std::move(consumer));
         return false;
       });
-  // For batch mode, the compiler must swallow diagnostics pertaining to
-  // non-primary files in order to avoid Xcode showing the same diagnostic
+  // For batch mode, the compiler must sometimes swallow diagnostics pertaining
+  // to non-primary files in order to avoid Xcode showing the same diagnostic
   // multiple times. So, create a diagnostic "eater" for those non-primary
   // files.
+  //
+  // This routine gets called for \c createJSONFixItDiagnosticConsumerIfNeeded
```

Comment:
This seems overly specific. "This routine gets called in cases where…" seems fine to me.

Reply:
Sure!